core/domain/user_query_domain.py

```diff
@@ -19,26 +19,85 @@
 from __future__ import annotations
 
 import collections
+import datetime
 
 from core import feconf
 from core import utils
 from core.constants import constants
 
-attribute_names = [ # pylint: disable=invalid-name
-    predicate['backend_attr'] for predicate in (
-        constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
-
-UserQueryParams = collections.namedtuple( # pylint: disable=invalid-name
-    'UserQueryParams', attribute_names, defaults=(None,) * len(attribute_names))
+from typing import TYPE_CHECKING, List, Optional
+
+attribute_names = [
+        predicate['backend_attr'] for predicate in (
+            constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
+
+# Check to see if list of attributes defined here (attribute_names_predefined)
+# is similar to the one we get during runtime (attribute_names)
+# from (/assets/constants.ts)
+if not TYPE_CHECKING:
+    attribute_names_predefined = [
+            'inactive_in_last_n_days',
+            'has_not_logged_in_for_n_days',
+            'created_at_least_n_exps',
+            'created_fewer_than_n_exps',
+            'edited_at_least_n_exps',
+            'edited_fewer_than_n_exps',
+            'created_collection'
+            ]
+
+    attribute_names_predefined.sort()
+    attribute_names.sort()
+
+    if len(attribute_names_predefined) != len(attribute_names):
+        raise utils.ValidationError(
+        'Attributes mismatch from attributes fetched from /assets/constants.ts')
+
+    if attribute_names_predefined != attribute_names:
+        raise utils.ValidationError(
+        'Attributes mismatch from attributes fetched from /assets/constants.ts')
+
```

Comment:
Thinking about it, can we verify this in backend tests?

Reply:
Yes , we can do that . Should i try to add one ?

Reply:
@vojtechjelinek i had some doubts ! Regarding validating this in backend tests . Should i just need to make a test file and add it to backend_test_shards.json . Will that suffice or i am missing something ?

Reply:
You can add the tet to the _user_query_domain_test.py_